SwiftDungeon/Enemy.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  Enemy.swift
+//  SwiftDungeon
+//
+//  Created by Toro Juan D. on 3/19/18.
+//  Copyright © 2018 Toro Juan D. All rights reserved.
+//
+
+import Foundation
+import SpriteKit
+
+class Enemy : Entity {
+    
+    //Speed of Enemy
+    private var velocity: CGFloat = 100
+    
+    // Position of target to chase
+    private var targetPosition: CGPoint = CGPoint(x: 0, y: 0)
+    
+    init() {
+        super.init(imageName: "slime_idle_01")
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func SetTarget(_ position: CGPoint) {
+        targetPosition = position
+    }
+    
+    func MoveToTarget() {
+        if(position.x > targetPosition.x) {
```

Comment:
Try using vector physics for 2D movement here instead of these if conditions. Get the direction you want to travel and you can do it all in one equation.